### ISSUE TITLE
Crash every time Signal is opened, after upgrading to 2.5.1

### ISSUE THREAD

**Issue by mruffalo** [WhisperSystems/Signal-iOS#1348]
- [x] I have searched open and closed issues for duplicates

----------------------------------------

### Bug description
Signal crashes immediately after opening it -- when I tap the icon, it expands into the blue splash screen with the iOS Signal logo in the center. Within half a second, the app exits and I am returned to my home screen. I never see the actual content of the app; straight from splash screen to iOS home screen.

### Steps to reproduce
- Open Signal

**Actual result:** Signal crashes and exits immediately
**Expected result:** Signal would not crash immediately and instead allow me to use the app

### Device info

**Device**: iPhone SE
**iOS version**: 10.0.2
**Signal version:** 2.5.1

### Link to debug log
Couldn't submit a debug log through Signal due to crashing, but I copied the iPhone diagnostic log to http://pastebin.com/r4cjK4v2 (and will email it to support@whispersystems.org as requested).

**Comment by michaelkirk** [comment]
=(

@mruffalo would you be interested in a debug build that will help identify the problem? email me at michael@whispersystems.org.

**Comment by mruffalo** [comment]
Sure. Email sent.

**Comment by klyonrad** [comment]
I can confirm this issue, have the Same behavior.

# Device Info:

**Device:** iPhone 6S
**iOS version:** 10.0.2
**Signal Version:** 2.5.1

**Comment by michaelkirk** [comment]
Same deal @klyonrad, if you're interested in a debug build, email me at michael@whispersystems.org.

Thanks!

**Comment by garrettr** [comment]
I have the same issue! It just started today after upgrading to the latest version of Signal for iOS.

**Device**: iPhone 5s
**iOS version**: 10.0.2
**Signal Version**: 2.5.1

I have sent an email about a debug build. Is there anything else I can do to help debug the issue?

**Comment by mmerickel** [comment]
I am having the same issue. The below log output from the device should make the issue clear for the developers.

```
Sep 26 16:56:55 My-iPhone Signal[12888] <Notice>: [TSStorageManager] Unarchiving key:+13107706742 from collection:TSRecipient and data <62706c69 73743030 d4010203 04050624 25582476 65727369 6f6e5824 6f626a65 63747359 24617263 68697665 72542474 6f701200 0186a0a8 07081112 16171e1f 55246e75 6c6cd409 0a0b0c0d 0e0f1057 64657669 63657358 756e6971 75654964 5f100f4d 544c4d6f 64656c56 65727369 6f6e5624 636c6173 73800380 06800280 071000d2 0c131415 5b4e532e 6f626a65 63742e30 80058004 1001d218 191a1b5a 24636c61 73736e61 6d655824 636c6173 7365735c 4e534f72 64657265 64536574 a21c1d5c 4e534f72 64657265 64536574 584e534f 626a6563 745c2b31 33313037 37303637 3432d218 1920215b 54535265 63697069 656e74a4 2022231d 5f101354 53596170 44617461 62617365 4f626a65 6374584d 544c4d6f 64656c5f 100f4e53 4b657965 64417263 68697665 72d12627 54726f6f 74800100 08001100 1a002300 2d003200 37004000 46004f00 57006000 72007900 7b007d00 7f008100 83008800 94009600 98009a00 9f00aa00 b300c000 c300d000 d900e600 eb00f700 fc011201 1b012d01 30013500 00000000 00020100 00000000 00002800 00000000 00000000 00000000 000137> failed with error: *** -[NSKeyedUnarchiver d
Sep 26 16:56:55 My-iPhone Signal[12888] <Notice>: [TSStorageManager] Raising exception since deserialization failed
Sep 26 16:56:55 My-iPhone Signal(CoreFoundation)[12888] <Notice>: *** Terminating app due to uncaught exception 'NSInvalidUnarchiveOperationException', reason: '*** -[NSKeyedUnarchiver decodeObjectForKey:]: cannot decode object of class (TSRecipient) for key (root); the class may be defined in source code or a library that is not linked'
*** First throw call stack:
```

Device: iPhone 6+
iOS version: 10.0.2
Signal Version: 2.5.1

**Comment by michaelkirk** [comment]
Thanks @mmerickel.

**Comment by UnrealAkama** [comment]
Signal is also showing this exact problem for me. 

Device: iPhone 6+
iOS version: 9.3.5
Signal Version: 2.5.1

I'm also willing to run a debugging build if you would like. In addition, is it safe to upgrade iOS or could that possibly corrupt something?

**Comment by mruffalo** [comment]
The debug build didn't crash, and I was able to get a debug log out of it. When I chose to submit it as a GitHub issue, it was automatically turned into a gist, which is available at https://gist.github.com/anonymous/abc64b27a8a0ed241d184fc01de86432. I'm not sure how useful it is, though.

**Comment by bwbroersma** [comment]
I can reproduce the issues too.
### Device info

**Device**: iPhone 5
**iOS version**: 10.1 Public beta 1 (14B55c)
**Signal version:** 2.5.1

**Comment by michaelkirk** [comment]
To anybody else that wants to try out a fix in the near time, email me your device UDID, and I can send you a debug build. http://whatsmyudid.com/

michael@whispersystems.org

**Comment by michaelkirk** [comment]
> I'm also willing to run a debugging build if you would like. In addition, is it safe to upgrade iOS or could that possibly corrupt something?

If you're experiencing the known bug, upgrading iOS won't do any damage as far as Signal is concerned. If you're experiencing some other, unknown bug, who can really say? =)

**Comment by michaelkirk** [comment]
Fix will go out in the next release, coming soon. Stay tuned.

**Comment by michaelkirk** [comment]
2.5.2 is available in the app store now.

**Comment by mmerickel** [comment]
Yep, I can confirm this fixed my issues. Thank you!

**Comment by stonie08** [comment]
I updated to 2.5.2, but the issue is *not* fixed for me. Anything I can do?

**Comment by michaelkirk** [comment]
@stonie08 If you have iTunes and are willing to install a debug build, send your (UDID)[http://whatsmyudid.com/] to michael@whispersystems.org

**Comment by klyonrad** [comment]
Stupid question: If I installed the debug build, can I not update to the App Store version anymore?

**Comment by bwbroersma** [comment]
2.5.2 fixed the crash on startup for me
### Device info

**Device**: iPhone 5
**iOS version**: 10.1 Public beta 1 (14B55c)
**Signal version:** 2.5.2

**Comment by rizwan95** [comment]
This problem exists in the latest version. 
Device info 

Device L iPhone 7 plus 
iOS Version : 12.1.4
Signal version : 2.36.0

I am attaching the crash report for your reference.

[Signal-2019-02-26-054426.txt](https://github.com/signalapp/Signal-iOS/files/2903201/Signal-2019-02-26-054426.txt)

**Comment by michaelkirk-signal** [comment]
@rizwan95 - This should be fixed in the upcoming 2.37.0 Beta - the beta isn't released yet, but if you'd like to sign up to be notified as soon as it is,  instructions on how to join the beta are here:

https://support.signal.org/hc/en-us/articles/360007318471-How-do-I-join-Signal-s-beta-

Also, in general, rather than commenting on a years-old closed issue, it's better to create a new one as the underlying cause is likely different.

**Comment by rizwan95** [comment]
Thanks for the update Michael. :)

I thought it might become a duplicate and hence I commented here. I will make sure that I create a new thread from now on depending upon the problem.
